src/uu/ls/src/ls.rs

```diff
@@ -3354,8 +3354,7 @@ fn color_name(
         // Use fn get_metadata_with_deref_opt instead of get_metadata() here because ls
         // should not exit with an err, if we are unable to obtain the target_metadata
         let md = get_metadata_with_deref_opt(target.p_buf.as_path(), path.must_dereference)
-            .unwrap_or_else(|_| target.get_metadata(out).unwrap().clone());
-
+            .unwrap_or_else(|_| path.p_buf.symlink_metadata().ok().unwrap().clone());
         apply_style_based_on_metadata(path, Some(&md), ls_colors, style_manager, &name)
```

Comment:
So I guess the bug was raised during the "unwrap"? In that case I have to ask the question: What do we do if `symlink_metadata` ever fails?

Also, is it even the right thing to do? Afterall, the metadata are to do presentation-stuff (color, hyperlinks, etc), so perhaps this might even be the right way to do it:

```suggestion
        let md = get_metadata_with_deref_opt(target.p_buf.as_path(), path.must_dereference);
        apply_style_based_on_metadata(path, md, ls_colors, style_manager, &name)
```

I don't know which approach is correct, please check against GNU 9.5.

Reply:
Yeah, so basically what happened is the ``target.get_metadata(out).unwrap().clone())`` called unwrap on a None. The ``get_metadata`` basically tries to get the metadata for the non-existent file in case of dangling symlinks. Panics  only happens we use a path to a file.  

The  ``symlink_metadata`` according to the docs  https://doc.rust-lang.org/std/fs/fn.symlink_metadata.html, only Errs in two cases, when path doesn't exist , which we already seem to handle elsewhere (tested it with a bunch of non-existent files) and the other seems to when we can't call a metadata on the file , I couldn't find a file where that is the case, would definitely appreciate some leads for it, so i can test.

But we can just remove unwraps all together with something like this .
 ```
        let md_res = get_metadata_with_deref_opt(target.p_buf.as_path(), path.must_dereference);
        let md = md_res.or(target.p_buf.symlink_metadata());

        apply_style_based_on_metadata(path, md.as_ref(),ok(), ls_colors, style_manager, &name)
```
We still need to call ``symlink_metadata()`` to color the file being pointed to by the color of the symlink itself.

Reply:
Thanks, this answers all my questions and worries :D

(Technically the file can still be missing due to TOCTOU, but that bug is literally everywhere, so I'll ignore it here.)